fetch_data.py

```diff
@@ -419,25 +439,109 @@ def create_csv(packaged_data, fieldnames, package_coords, fill_di=None):
                 row_di[fieldname] = value
         # write the actual value
         row_di["value"] = get_from_dict(packaged_data, coords)
-        writer.writerow(row_di)
-    return output.getvalue()
+        rows.append(row_di)
+    return rows
 
 
-def return_csv(csv_data):
-    """Return the CSV data as a download
+def write_csv(csv_dicts, fieldnames, filename, metadata=None):
```

Comment:
Nice, this all looks good and makes sense to divy it up among the new functions this way.